rest-api/src/main/scala/za/co/absa/enceladus/rest_api/controllers/v3/VersionedModelControllerV3.scala

```diff
@@ -148,25 +156,42 @@ abstract class VersionedModelControllerV3[C <: VersionedModel with Product
     } else if (version != item.version) {
       Future.failed(new IllegalArgumentException(s"URL and payload version mismatch: ${version} != ${item.version}"))
     } else {
-      versionedModelService.update(user.getUsername, item).map {
-        case Some((updatedEntity, validation)) =>
-          createdWithNameVersionLocationBuilder(updatedEntity.name, updatedEntity.version, request, stripLastSegments = 2).body(validation)
-        case None => throw notFound()
+      versionedModelService.isDisabled(item.name).flatMap { isDisabled =>
+        if (isDisabled) {
+          throw EntityDisabledException(s"Entity ${item.name} is disabled. Enable it first to create new versions.")
+        } else {
+          versionedModelService.update(user.getUsername, item).map {
+            case Some((updatedEntity, validation)) =>
+              createdWithNameVersionLocationBuilder(updatedEntity.name, updatedEntity.version, request, stripLastSegments = 2).body(validation)
+            case None => throw notFound()
+          }
+        }
       }
     }
   }
 
-  @DeleteMapping(Array("/{name}", "/{name}/{version}"))
+  @PutMapping(Array("/{name}"))
   @ResponseStatus(HttpStatus.OK)
-  def disable(@PathVariable name: String,
-              @PathVariable version: Optional[String]): CompletableFuture[UpdateResult] = {
-    val v = if (version.isPresent) {
-      // For some reason Spring reads the Optional[Int] param as a Optional[String] and then throws ClassCastException
-      Some(version.get.toInt)
-    } else {
-      None
+  def enable(@PathVariable name: String): CompletableFuture[DisabledPayload] = {
```

Comment:
Since we block the disable if the entity is used. Why are we not blocking re-enables if the entity is missing schema.

Step by step:
1. Disable datasetX that uses schemaX
2. Disable schemaX 
3. Enable datasetX passes. 

I would expect it to fail

Reply:
Nice catch! Yes, this should be probably checked, too. Thank you.

Reply:
This will not be trivial -> https://github.com/AbsaOSS/enceladus/issues/2065